src/stp/new_tactics/ChipAtPos.cpp

```diff
@@ -0,0 +1,107 @@
+//
+// Created by timovdk on 3/13/20.
+//
+
+#include "stp/new_tactics/ChipAtPos.h"
+
+#include <roboteam_utils/Print.h>
+#include <stp/new_skills/Chip.h>
+#include <stp/new_skills/Rotate.h>
+
+namespace rtt::ai::stp::tactic {
+
+ChipAtPos::ChipAtPos() {
+    // Create state machine of skills and initialize first skill
+    skills = rtt::collections::state_machine<Skill, Status, StpInfo>{skill::Rotate(), skill::Chip()};
+    skills.initialize();
+}
+
+void ChipAtPos::onInitialize() noexcept {}
+
+void ChipAtPos::onUpdate(Status const &status) noexcept {}
+
+void ChipAtPos::onTerminate() noexcept {
+    // Call terminate on all skills
+    for (auto &x : skills) {
+        x->terminate();
+    }
+}
+
+StpInfo ChipAtPos::calculateInfoForSkill(StpInfo const &info) noexcept {
+    if (!info.getBall() || !info.getRobot() || !info.getField()) {
+        RTT_WARNING("No Ball, Robot or Field present in StpInfo");
+        return {};
+    }
```

Comment:
Present in base class already. Check development

Reply:
woops forgot to remove this, added this before the meeting this morning